src/main.rs

```diff
@@ -179,6 +179,9 @@ fn get_pants_process() -> Result<Process> {
         ));
     }
     if let Some(version) = pants_version {
+        if delegate_bootstrap {
+            env.push(("_PANTS_OVERRIDE_VERSION".into(), version.clone().into()));
```

Comment:
I think it would be good to follow up on the ick by adding comments to this env var over in Pants (as well as the build root override env var) to point out they are being used externally here; i.e.: don't yank or rename without consultation / coordination.

Reply:
Good call. Will do. 👍🏽

Reply:
pantsbuild/pants#18170